block.go

```diff
@@ -40,24 +34,18 @@ func (b *Block) GetID() string {
 	return fmt.Sprintf("%x", b.ID)
 }
 
-func (b Block) Marshal() ([]byte, error) {
-	buf := bytes.NewBuffer(make([]byte, 0, 8+SizeMerkleNodeID+4+4+len(b.Transactions)*SizeTransactionID))
-
-	if err := binary.Write(buf, binary.BigEndian, b.Index); err != nil {
-		return nil, errors.Wrap(err, "error marshaling index")
-	}
-
-	buf.Write(b.Merkle[:])
+func (b Block) Marshal() []byte {
+	buf := make([]byte, 8+SizeMerkleNodeID+4+len(b.Transactions)*SizeTransactionID)
 
-	if err := binary.Write(buf, binary.BigEndian, uint32(len(b.Transactions))); err != nil {
-		return nil, errors.Wrap(err, "error marshaling transactions num")
-	}
+	binary.BigEndian.PutUint64(buf[0:8], b.Index)
+	copy(buf[8:8+SizeMerkleNodeID], b.Merkle[:])
+	binary.BigEndian.PutUint32(buf[8+SizeMerkleNodeID:8+SizeMerkleNodeID+4], uint32(len(b.Transactions)))
 
-	for _, id := range b.Transactions {
-		buf.Write(id[:])
+	for i, id := range b.Transactions {
+		copy(buf[8+SizeMerkleNodeID+4+i*SizeTransactionID:8+SizeMerkleNodeID+4+i*SizeTransactionID+SizeTransactionID], id[:])
```

Comment:
maybe it makes sense to create separate variables for parts of slice indexes? like
```
prefix := 8+SizeMerkleNodeID+4
for i, id := range b.Transactions {
    copy(buf[prefix+i*SizeTransactionID:prefix+(i+1)*SizeTransactionID]
}
```

Reply:
I've already made that changes.